.github/workflows/publish-command.yml

```diff
@@ -65,10 +153,32 @@ jobs:
           aws-access-key-id: ${{ secrets.SELF_RUNNER_AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
           github-token: ${{ needs.find_valid_pat.outputs.pat }}
+  preprocess-matrix:
+    needs: start-publish-image-runner-0
+    runs-on: ${{ needs.start-publish-image-runner-0.outputs.label }}
+    outputs:
+      connectorjson: ${{ steps.preprocess.outputs.connectorjson }}
+    steps:
+      - id: preprocess
+        run: |
+          start="[\""
+          replace="\",\""
+          end="\"]"
+          stripped_connector="$(echo "${{ github.event.inputs.connector }}" | tr -d ' ')"
+          middle=${stripped_connector//,/$replace}
+          full="$start$middle$end"
+          echo "::set-output name=connectorjson::$full"
   publish-image:
     timeout-minutes: 240
-    needs: start-publish-image-runner
-    runs-on: ${{ needs.start-publish-image-runner.outputs.label }}
+    needs:
+      - start-publish-image-runner-0
+      - preprocess-matrix
+    strategy:
+      max-parallel: 5
+      fail-fast: false
+      matrix:
+        connector: ${{ fromJSON(needs.preprocess-matrix.outputs.connectorjson) }}
+    runs-on: self-hosted # TODO: need to change this to run on custom label when I can implement that
```

Comment:
this is very bad as its picking up any self-hosted agent. I need to make a change [in this repo](https://github.com/supertopher/ec2-github-runner) we're using so that I can add a custom group label to the agents spun up in this job so we can use them as a pool.